### ISSUE TITLE
[Feature]: Add templates to actions

### ISSUE THREAD

**Issue by SteffenM-DE** [piitaya/lovelace-mushroom#895]
### Requirements

- [X] I have updated Mushroom to the latest available version
- [X] I did a search to see if there is a similar issue or if a pull request is open.

### Is your feature request related to a problem?

Often I want to have different actions on the state of a entity but there is no templates allowed at the actions. So it would be nice if actions and the action confirmation could contain templates.

### Describe the solution you'd like

Normal templates support at tap_action, hold_action and double_tap_action support for the template card and the template chip and the confirmation (for example if state on 'Do you want to switch off' and if state is off 'Do you want switch on').

### Describe alternatives you've considered

Tryed many thinks, but only other custom cards could do that.

**Comment by timdonovanuk** [comment]
Wow, I think I was about to raise the same issue. 

This should work:

```
type: custom:mushroom-chips-card
chips:
  - type: template
    entity: climate
    content: Boost Bedroom Heating
    tap_action:
      action: call-service
      service: climate.set_temperature
      data:
        temperature: >
          {{ (state_attr('climate.bedroom', 'temperature')|float)}}
        target:
          entity_id: climate.bedroom

```
It lets me save the card, but the service call fails because I think templates don't resolve properly in actions. I get an error `Failed to call service climate/set_temperature. expected float for dictionary value @ data['temperature']`

**Comment by piitaya** [comment]
Actions are not allowed in template (it's a HA limitation, not mushroom one). You can use a script as a workaround.

**Comment by SteffenM-DE** [comment]
Ok thanks, a script is a solution I never think about. So I could use the action: call-service to handle different thinks. Only the confirmation could not be variable.

I think the custom button-card could do that all but it is a little special using Javascript as templates.

**Comment by piitaya** [comment]
Yes, Mushroom use the HA built-in actions editor.

**Comment by MatteGary** [comment]
I almost opened the same issue.
@piitaya you mentioned that "Templates are not allowed in actions (it's a HA limitation, not mushroom one).", but I built a custom button-card using this snippet for the tap_action

    tap_action:
      action: url
      url_path: '[[[ return `http://${entity.attributes.ip_address}` ]]]'

and it works. Should it work also in Mushroom?

**Comment by piitaya** [comment]
This template format is a custom thing from custom button card. Official HA cards and Mushroom cards don't support these templates.

**Comment by MatteGary** [comment]
Oh ok, didn't know that.
I guess I have to hard code the ip in the UI config.
Thanks

**Comment by sstratoti** [comment]
What about adding templating actions to just the templating cards?
Or the ability to switch between native HA built-in actions and a template editor for the actions section in the Card's UI?

I'm going to try to make a case here. In my specific use case, I'm defining the action and the entity/URL in my custom button card based on the attributes of a dynamic entity.

For example, my sensor.active_alerts has the following attributes:
```
name: HACS Update Available 
label:  
backgroundcolor: #66b2ff 
icon: mdi:archive-arrow-down-outline 
id: HACS Update Available 
textcolor: white 
alertcount: 2 
animation: blink 3s ease infinite 
entity:  
internalurl: /hacs/entry 
externalurl:  
friendly_name: Active Alerts
```
After a few seconds, node-red updates the attributes of the entity with new data:
```
name: Roborock Vacuum 
label: Unavailable - Check Dock 
backgroundcolor: #ff6666 
icon: mdi:robot-vacuum 
id: Roborock Unavailable 
textcolor: black 
alertcount: 2 
animation: none 
entity:  
internalurl:  
externalurl:  
friendly_name: Active Alerts
```

Unfortunately I can't set the URL target dynamically in the mushroom-template-card. So if I get a camera alert that swaps in, if I press the button - I'd like to head over to my camera page. If its the HACS alert, send me to HACS. Etc. 

Or, if my door is unlocked, and an alert is displayed - it says "press me to lock it" and toggles the lock service call.

Based on an entities state/attributes, you may want the actions of the button to behave in different ways.

Example of the custom-button-card I've created:

![2023-04-14_23-52-22 (1)](https://user-images.githubusercontent.com/1392597/232181657-5cbac9ea-e9c5-419c-94b5-09e1574cada8.gif)

**Comment by piitaya** [comment]
I close it because Mushroom can not add this support because it use built-in HA actions and they do not supports templating.

**Comment by dinhani** [comment]
I am using the Chips components and spent two days trying to find a workaround for generating a dynamic URL when clicking a button. So I just want to leave registered the solution I found for others. 

I achieved a solution using `browser_mod` with a custom script to perform the navigation.

In the component I use `perform-action` passing all static data I have:

```yaml
action: perform-action
perform_action: script.my_custom_navigation_script
data:
  param1: static-value-1
  param2: static-value-2
```

And inside the script I use `browser_mod.navigate` to go the desired page. The script has access to all necessary state to generate the desired URL.

cc: https://github.com/piitaya/lovelace-mushroom/issues/1544
